### PR TITLE
oic: Improve query handling in oic server and client

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/lib/comms/sol-oic-server.c [hunk 1]
```diff
@@ -183,12 +183,41 @@ static const struct sol_oic_resource_type oic_p_resource_type = {
...
+
+static bool
+oic_query_split(struct sol_str_slice query, struct sol_str_slice *key, struct sol_str_slice *value)
+{
+    char *sep;
+
+    sep = memchr(query.data, '=', query.len);
+    SOL_NULL_CHECK(sep, false);
+    *key = SOL_STR_SLICE_STR(query.data, sep - query.data);
+    *value = SOL_STR_SLICE_STR(sep + 1, query.len - (sep - query.data + 1));
```
Comment (vcgomes) [inline src/lib/comms/sol-oic-server.c]: The case that the received query is something like `if=` could cause problems, no?

Comment (otaviobp) [inline src/lib/comms/sol-oic-server.c]: No, it won't fail. value->len will be 0, so it will be ignored.

Comment (vcgomes) [inline src/lib/comms/sol-oic-server.c]: Cool. So please make `sep` const. (I really don't like these libc functions that drop the const)

**Comment by otaviobp** [comment]
merged

**Pull request closed without merging**